### PR TITLE
Complete  the unittests for the `POVMPostProcessor` class

### PR DESCRIPTION
**Pull request by timmintam**
As part of #37.

### CODE REVIEW

File: test/post_processor/test_post_processor.py [hunk 1]
```diff
@@ -8,42 +8,157 @@
...
 
 
 class TestPostProcessor(TestCase):
-    """TODO."""
+    """Test the methods and attributes of the :class:`.POVMPostProcessor class`."""
 
-    def test_init(self):
-        """TODO."""
+    def __init__(self, methodName: str = "runTest") -> None:
+        super().__init__(methodName)
```
Comment (mrossinek) [inline test/post_processor/test_post_processor.py:30]: Why are these changes necessary?

File: test/post_processor/test_post_processor.py [hunk 2]
```diff
@@ -8,42 +8,157 @@
...
         qc = QuantumCircuit(2)
         qc.h(0)
         qc.cx(0, 1)
 
-        sampler = Sampler()
+        sampler = Sampler(seed=1)
         povm_sampler = POVMSampler(sampler=sampler)
 
-        measurement = ClassicalShadows(num_qubits=2)
+        self.measurement = ClassicalShadows(num_qubits=2, seed_rng=2)
```
Comment (mrossinek) [inline test/post_processor/test_post_processor.py:39]: Could you set a single seed as a variable (maybe even on the class) and use its value for all seeds?
That makes it a bit easier to see where seeds are being used.
For example:
```python
class TestPostProcessor:

    RNG_SEED = 42

    def __init__(self):
        sampler = Sampler(seed=self.RNG_SEED)

# and so on...
```

File: test/post_processor/test_post_processor.py [hunk 3]
```diff
@@ -8,42 +8,157 @@
...
         result = job.result()
-        pub_result = result[0]
+        self.pub_result = result[0]
+
+    def test_init(self):
+        """Test that ``__init__`` works correctly."""
+        with self.subTest("Initialization with default ``dual``."):
+            post_processor = POVMPostProcessor(self.pub_result)
+            self.assertIsInstance(post_processor, POVMPostProcessor)
+            self.assertIsNone(post_processor._dual)
```
Comment (mrossinek) [inline test/post_processor/test_post_processor.py:50]: Normally we would not test private attributes but here I agree with it because the `_dual` is the internal state of the `dual` property. And calling `post_processor.dual` would actually build a Dual out...

File: test/post_processor/test_post_processor.py [hunk 4]
```diff
@@ -8,42 +8,157 @@
...
+        observable = SparsePauliOp(["ZZ", "XX", "YY"], coeffs=[1, 2, 3])
+        post_processor = POVMPostProcessor(self.pub_result)
+        with self.subTest("Test with default ``loc`` for un-parametrized circuit."):
+            exp_val, std = post_processor.get_expectation_value(observable)
+            self.assertAlmostEqual(exp_val, -0.6328124999999991)
+            self.assertAlmostEqual(std, 0.7276646052978725)
+        with self.subTest("Test with specified ``loc`` argument."):
+            exp_val, std = post_processor.get_expectation_value(observable, 0)
+            self.assertAlmostEqual(exp_val, -0.6328124999999991)
+            self.assertAlmostEqual(std, 0.7276646052978725)
```
Comment (mrossinek) [inline test/post_processor/test_post_processor.py:120]: Can we change this slightly such that these two test cases don't have the exact same result?

File: test/post_processor/test_post_processor.py [hunk 5]
```diff
@@ -8,42 +8,157 @@
...
+                    std,
+                    np.array(
+                        [[0.77141384, 0.76831946, 0.57462236], [0.78996484, 0.73757516, 0.55792747]]
+                    ),
+                )
+            )
 
-        self.assertIsInstance(exp_val, float)
-        self.assertIsInstance(std, float)
+    def test_single_exp_value_and_std(self):
```
Comment (mrossinek) [inline test/post_processor/test_post_processor.py:158]: This is testing a private method but since this one is kind of crucial and unlikely to change in the future (and since we have this test), let's leave it as is :+1:

**Review by mrossinek — COMMENTED** [review]
Nice! Some comments, but overall this is looking good already :+1:

File: test/post_processor/test_post_processor.py (hunk 1 above)
Comment (timmintam) [reply test/post_processor/test_post_processor.py:30]: I added an `__init__` to the test to define `self.pub_result` which is re-used multiple times in different tests. This is to avoid having to define a circuit, sampler, etc and run a job for each test in the file.

Comment (mrossinek) [reply test/post_processor/test_post_processor.py:30]: Yes, having the `__init__` is fine. I have just not seen this `methodName` before and am wondering why/if it is relevant/needed

Comment (timmintam) [reply test/post_processor/test_post_processor.py:32]: I don't remember where I saw I was supposed to add the argument `methodName`, but I get an error if I remove it.
I could also write 
```python
def __init__(self, *args, **kwargs) -> None:
    super().__init__(*args, **kwargs)
```
alternatively, it also works. What do you think ?

Comment (mrossinek) [reply test/post_processor/test_post_processor.py:32]: Ah I remember again! We normally leave `__init__` unchanged and instead overwrite the `setUp` method (https://docs.python.org/3/library/unittest.html#unittest.TestCase.setUp).
So you would do:
```python
class TestClass(unittest.TestCase):
    def setUp(self) -> None:
        super().setUp()

        # whatever else you want to do
```

Comment (timmintam) [reply test/post_processor/test_post_processor.py:32]: Okay I see, I'll make the change!

**Comment by timmintam** [comment]
I accidentally added the commit [bc19ab8](https://github.com/qiskit-community/povm-toolbox/pull/41/commits/bc19ab8d85c4b783901cad35106f4fed8b22cd39) in this branch

**Review by mrossinek — APPROVED** [review]

**Comment by mrossinek** [comment]
> I accidentally pushed the commit [bc19ab8](https://github.com/qiskit-community/povm-toolbox/pull/41/commits/bc19ab8d85c4b783901cad35106f4fed8b22cd39) in this branch

Yes I saw. That's okay :+1:
